code/workspaces/web-app/src/reducers/replaceProjectCategoryItems.js

```diff
@@ -0,0 +1,12 @@
+import { getCategoryFromTypeName } from 'common/src/stackTypes';
+
+export default function replaceProjectCategoryItems(value, payload) {
+  if (!payload || payload.length === 0) return [...value];
+  if (!value || value.length === 0) return [...payload];
+  const { projectKey, type } = payload[0];
+  const category = getCategoryFromTypeName(type);
+  return [
+    ...value
+      .filter(item => item.projectKey !== projectKey || getCategoryFromTypeName(item.type) !== category),
+    ...payload];
```

Comment:
I find this a bit easier to read when spaced out differently. I don't know what you think?

```js
  return [
    ...value.filter(
      item => item.projectKey !== projectKey || getCategoryFromTypeName(item.type) !== category
    ),
    ...payload,
  ];
```

Reply:
Given there's not a series of filters, yes - fixed